src/core/dbus-service.c

```diff
@@ -218,6 +219,72 @@ int bus_service_method_mount_image(sd_bus_message *message, void *userdata, sd_b
         return bus_service_method_mount(message, userdata, error, true);
 }
 
+int bus_service_method_dump_file_descriptor_store(sd_bus_message *message, void *userdata, sd_bus_error *error) {
+        _cleanup_(sd_bus_message_unrefp) sd_bus_message *reply = NULL;
+        Service *s = ASSERT_PTR(userdata);
+        int r;
+
+        assert(message);
+
+        r = mac_selinux_unit_access_check(UNIT(s), message, "status", error);
+        if (r < 0)
+                return r;
+
+        if (s->n_fd_store_max == 0 && s->n_fd_store == 0)
+                return sd_bus_error_setf(error, SD_BUS_ERROR_INVALID_ARGS, "File descriptor store not enabled for %s.", UNIT(s)->id);
```

Comment:
Should this simply return an empty array? After all it's not really an error, it's just that there's no info to show

Reply:
we return a empty array if the fdstore is enabled but nothing in there. I think there's value in allowing clients to distinguish the cases "fdstore disabled" from "fdstore enabled but empty". The former gets this error, the latter an empty array.

Note the check checks if the limit and fill level is both zero. Only if both are zero we'll return the error. this is to cover cases where a service is started with fdstore on, receives a bunch of fds, and then a daemon reload is issued and the unit changed to disallow an fdstore. In such a case we should still show the fdstore that was acquired, even if now we wouldn't allow that anymore.

Reply:
I'm not sure - "invalid arguments" to be sounds like the input was bogus, but it is clearly not the case here, it's simply that an option was disabled. When receiving `SD_BUS_ERROR_INVALID_ARGS`, my immediate reaction is to suspect there's a coding error in the client and it's sending bogus inputs.